src/any_agent/tracing/exporter.py

```diff
@@ -89,7 +89,7 @@ def export(self, spans: Sequence["ReadableSpan"]) -> SpanExportResult:  # noqa:
                 if span_kind == "LLM" and self.tracing_config.cost_info:
                     span.add_cost_info()
 
-                self.traces[trace_id].spans.append(span)
```

Comment:
calling add_span then lets us handle cache invalidation if someone is calling the total_cost property multiple times before the trace is finished (this is an extremely unlikely scenario as far as I can tell)